lib/lastfm/user.rb

```diff
@@ -4,41 +4,17 @@ module Lastfm
   class User
     def initialize(username)
       @username = username
+      @http_client = HttpClient.new
     end
 
     def recent_tracks(period, page_number = 1)
-      RecentTrackList.new(response_body(
+      @http_client.get(
+        RecentTrackList,
+        user: @username,
```

Comment:
I'm not a big fan of letting the http client build the object. I'd rather let it handle just HTTP stuff, and move the instantiation here:

```suggestion
      RecentTrackList.new(@http_client.get(
        user: @username,
```

Reply:
I could go either way on this one. There may be another level of abstraction that I need to add. At some point, I want this method to return a collection of (yet undetermined) things. I want to give the HTTP client an object that will parse a specific endpoint. There should then be another object to take that parsed data and return it to something for the library.